### PR TITLE
Generate again conda package for walking-controllers and fix conda metapackages generation

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/robotology-superbuild/issues/1577 . 

In a rush to actually generate conda packages and fix https://github.com/robotology/robotology-superbuild/issues/1572, we disabled walking-controllers generation, as it was failing on Linux. However, the generation was failing as the latest bipedal-locomotion-framework was compiled against ipopt 3.14.14, while icub-main was still building with 3.14.13 . This was happening as:
* the ipopt 3.14.14 migration was still ongoing (https://github.com/conda-forge/ipyopt-feedstock/pull/29, https://github.com/conda-forge/cyipopt-feedstock/pull/52 and https://github.com/conda-forge/biorbd-feedstock/pull/110 were missing)
* bipedal-locomotion-framework 0.18.0 was released after blf feedstock was migrated from ipopt 3.14.13 to ipopt 3.14.14, so for 0.18.0 no ipopt 3.14.13 build was available (see https://github.com/conda-forge/bipedal-locomotion-framework-feedstock/pull/51 and https://github.com/conda-forge/bipedal-locomotion-framework-feedstock/pull/53)
* icub-main compiled by the robotology conda packages CI was compiled against ipopt 3.14.13, as we currently build packages against the version pinned in conda-forge-pinning, see https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/537bce44ea1587c63d19d3f7b545fb46e1af5b42/recipe/conda_build_config.yaml#L409 and https://github.com/robotology/robotology-superbuild/blob/d5c49b8495bdb48ac8f6f29115977f80e0413c7f/.github/workflows/generate-conda-packages.yaml#L179

We fixed this in https://github.com/robotology/robotology-superbuild/pull/1578 by manually pinning ipopt to 3.14.14 in the robotology conda packages generation CI. This can be removed once ipopt 3.14.14 abi migration is completed at the conda-forge level.

To understand how to avoid this in the future:
* Just avoid hacks like removing the walking-controllers build. The CI did its work of reporting that there was something wrong, and instead of fixing it we tried to hack, and this is we ended up with the problem in the first place.
* Probably at this point we can consider moving icub-main (and icub-firmware-shared) to conda-forge. I will open a dedicated issue to discuss that, but if we had that, icub-main probably would have been migrated to ipopt by conda-forge's migration mechanism, and we would not had this ipopt-related incompatibility in the first place.
* Something like https://github.com/robotology/robotology-superbuild/issues/1584 could also be useful.

cc @GiulioRomualdi

### CODE REVIEW

**Comment by traversaro** [comment]
`generate-conda-packages`  test build with these modifications : https://github.com/robotology/robotology-superbuild/actions/runs/7693316196

**Comment by traversaro** [comment]
robotology-distro-all generation still fails:
~~~
2024-01-29T11:06:18.6974119Z ####################################################################################
2024-01-29T11:06:18.6974492Z Resource usage summary:
2024-01-29T11:06:18.6974672Z 
2024-01-29T11:06:18.6974761Z Total time: 0:01:38.8
2024-01-29T11:06:18.6975037Z CPU usage: sys=0:00:00.0, user=0:00:00.0
2024-01-29T11:06:18.6975371Z Maximum memory usage observed: 0B
2024-01-29T11:06:18.6975741Z Total disk usage observed (not including envs): 0B
2024-01-29T11:06:26.2194124Z WARNING: No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.22
2024-01-29T11:06:26.2638295Z Updating build index: /usr/share/miniconda3/envs/test/conda-bld
2024-01-29T11:06:26.2638861Z 
2024-01-29T11:06:26.2639046Z Adding in variants from internal_defaults
2024-01-29T11:06:26.2639746Z Adding in variants from /usr/share/miniconda3/envs/test/conda_build_config.yaml
2024-01-29T11:06:26.2640967Z Adding in variants from /home/runner/work/robotology-superbuild/robotology-superbuild/conda/conda_build_config.yml
2024-01-29T11:06:27.8024170Z Attempting to finalize metadata for robotology-distro-all
2024-01-29T11:06:28.3831280Z warning  libmamba Cache file "/home/runner/conda_pkgs_dir/cache/497deca9.json" was modified by another program
2024-01-29T11:06:28.3834939Z warning  libmamba Cache file "/home/runner/conda_pkgs_dir/cache/09cdf8bf.json" was modified by another program
2024-01-29T11:06:28.3837565Z warning  libmamba Cache file "/home/runner/conda_pkgs_dir/cache/c47980bd.json" was modified by another program
2024-01-29T11:06:29.0720259Z warning  libmamba Cache file "/home/runner/conda_pkgs_dir/cache/ebeb5136.json" was modified by another program
2024-01-29T11:06:36.4210079Z Reloading output folder: /usr/share/miniconda3/envs/test/conda-bld
2024-01-29T11:06:36.5543959Z Reloading output folder: /usr/share/miniconda3/envs/test/conda-bld
2024-01-29T11:06:36.7206817Z BUILD START: ['robotology-distro-all-None-ha770c72_112.tar.bz2']
2024-01-29T11:06:36.7208036Z Reloading output folder: /usr/share/miniconda3/envs/test/conda-bld
2024-01-29T11:06:36.8599331Z Reloading output folder: /usr/share/miniconda3/envs/test/conda-bld
2024-01-29T11:06:39.7673006Z Traceback (most recent call last):
2024-01-29T11:06:39.7674003Z Mamba failed to solve:
2024-01-29T11:06:39.7674720Z  - openvr 1.16.8.1.*
2024-01-29T11:06:39.7675320Z  - libosqp 0.6.3.*
2024-01-29T11:06:39.7675870Z  - speech 1.2.0.*
2024-01-29T11:06:39.7676250Z  - yarp-device-xsensmt 0.3.0.*
2024-01-29T11:06:39.7676929Z  - yarp-openvr-trackers 0.0.1.*
2024-01-29T11:06:39.7677502Z  - robots-configuration 2.5.2.*
2024-01-29T11:06:39.7677978Z  - icub-tests 1.25.0.*
2024-01-29T11:06:39.7678421Z  - wb-toolbox 5.6.0.*
2024-01-29T11:06:39.7678858Z  - icub-models 2.4.1.*
2024-01-29T11:06:39.7679271Z  - yarp-devices-forcetorque 0.3.1.*
2024-01-29T11:06:39.7679796Z  - bipedal-locomotion-framework 0.18.0.*
2024-01-29T11:06:39.7680321Z  - walking-teleoperation 1.3.4.*
2024-01-29T11:06:39.7680870Z  - icub-main 2.4.1.*
2024-01-29T11:06:39.7681433Z  - yarp-matlab-bindings 3.9.0.*
2024-01-29T11:06:39.7681914Z  - yarp-device-openxrheadset 0.0.3.*
2024-01-29T11:06:39.7682414Z  - blocktest-yarp-plugins 1.1.4.*
2024-01-29T11:06:39.7683074Z  - robot-testing-framework 2.0.1.*
2024-01-29T11:06:39.7683551Z  - icub-firmware-shared 1.37.0.*
2024-01-29T11:06:39.7683898Z  - openxr-sdk 1.0.33.*
2024-01-29T11:06:39.7684205Z  - whole-body-controllers 2.5.6.*
2024-01-29T11:06:39.7684529Z  - wearables 1.8.0.*
2024-01-29T11:06:39.7684814Z  - idyntree-yarp-tools 0.0.9.*
2024-01-29T11:06:39.7685158Z  - liblie-group-controllers 0.2.0.*
2024-01-29T11:06:39.7685608Z  - pyqtconsole 1.2.3.*
2024-01-29T11:06:39.7686025Z  - human-gazebo 1.1.0.*
2024-01-29T11:06:39.7686308Z  - qpoases 3.2.1.*
2024-01-29T11:06:39.7686591Z  - walking-controllers 0.8.0.*
2024-01-29T11:06:39.7686922Z  - ergocub-software 0.6.0.*
2024-01-29T11:06:39.7687251Z  - robot-log-visualizer 0.8.0.*
2024-01-29T11:06:39.7687579Z  - blockfactory 0.8.4.*
2024-01-29T11:06:39.7687906Z  - idyntree-matlab-bindings 10.2.0.*
2024-01-29T11:06:39.7688245Z  - manif 0.0.4.*
2024-01-29T11:06:39.7688645Z  - meshcat-python 0.3.2.*
2024-01-29T11:06:39.7689006Z  - libyarp 3.9.0.*
2024-01-29T11:06:39.7689273Z  - osqp-eigen 0.8.1.*
2024-01-29T11:06:39.7689570Z  - icub-contrib-common 1.19.0.*
2024-01-29T11:06:39.7689933Z  - matlab-whole-body-simulator 3.3.0.*
2024-01-29T11:06:39.7690306Z  - whole-body-estimators 0.10.0.*
2024-01-29T11:06:39.7690633Z  - idyntree 10.2.0.*
2024-01-29T11:06:39.7690931Z  - casadi-matlab-bindings 3.6.3.0.*
2024-01-29T11:06:39.7691379Z  - icub-basic-demos 1.24.0.*
2024-01-29T11:06:39.7691740Z  - funny-things 2.2.0.*
2024-01-29T11:06:39.7692140Z  - libbayes-filters-lib 0.10.0.*
2024-01-29T11:06:39.7692472Z  - librobometry 1.2.2.*
2024-01-29T11:06:39.7692823Z  - libunicycle-footstep-planner 0.7.0.*
2024-01-29T11:06:39.7693297Z  - proxsuite 0.6.0.*
2024-01-29T11:06:39.7693669Z  - human-dynamics-estimation 3.0.0.*
2024-01-29T11:06:39.7694257Z  - event-driven 2.1.*
2024-01-29T11:06:39.7694728Z  - blocktest 2.3.7.*
2024-01-29T11:06:39.7695086Z  - libgazebo-yarp-plugins 4.9.0.*
2024-01-29T11:06:39.7695410Z  - casadi 3.6.3.*
2024-01-29T11:06:39.7695678Z  - osqp-matlab 0.6.2.3.*
2024-01-29T11:06:39.7695975Z  - cppad 20230000.0.*
2024-01-29T11:06:39.7696279Z  - resolve-robotics-uri-py 0.1.0.*
2024-01-29T11:06:39.7696627Z  - ycm-cmake-modules 0.16.2.*
2024-01-29T11:06:39.7696945Z  - libmatio-cpp 0.2.3.*
2024-01-29T11:06:39.7697114Z 
2024-01-29T11:06:39.7697205Z with channels:
2024-01-29T11:06:39.7697441Z  - local
2024-01-29T11:06:39.7697670Z  - conda-forge
2024-01-29T11:06:39.7698083Z  - robotology
2024-01-29T11:06:39.7698223Z 
2024-01-29T11:06:39.7698326Z The reported errors are:
2024-01-29T11:06:39.7698660Z - Encountered problems while solving:
2024-01-29T11:06:39.7699614Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/boa/cli/mambabuild.py", line 141, in mamba_get_install_actions
2024-01-29T11:06:39.7700452Z     solution = solver.solve_for_action(_specs, prefix)
2024-01-29T11:06:39.7700859Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-01-29T11:06:39.7701671Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/boa/core/solver.py", line 242, in solve_for_action
2024-01-29T11:06:39.7702421Z     t = self.solve(specs)
2024-01-29T11:06:39.7702675Z         ^^^^^^^^^^^^^^^^^
2024-01-29T11:06:39.7703347Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/boa/core/solver.py", line 232, in solve
2024-01-29T11:06:39.7704138Z     raise RuntimeError("Solver could not find solution." + error_string)
2024-01-29T11:06:39.7704732Z RuntimeError: Solver could not find solution.Mamba failed to solve:
2024-01-29T11:06:39.7705201Z  - openvr 1.16.8.1.*
2024-01-29T11:06:39.7705476Z  - libosqp 0.6.3.*
2024-01-29T11:06:39.7705739Z  - speech 1.2.0.*
2024-01-29T11:06:39.7706020Z  - yarp-device-xsensmt 0.3.0.*
2024-01-29T11:06:39.7706361Z  - yarp-openvr-trackers 0.0.1.*
2024-01-29T11:06:39.7706703Z  - robots-configuration 2.5.2.*
2024-01-29T11:06:39.7707093Z  - icub-tests 1.25.0.*
2024-01-29T11:06:39.7707370Z  - wb-toolbox 5.6.0.*
2024-01-29T11:06:39.7707651Z  - icub-models 2.4.1.*
2024-01-29T11:06:39.7707968Z  - yarp-devices-forcetorque 0.3.1.*
2024-01-29T11:06:39.7708365Z  - bipedal-locomotion-framework 0.18.0.*
2024-01-29T11:06:39.7708757Z  - walking-teleoperation 1.3.4.*
2024-01-29T11:06:39.7709086Z  - icub-main 2.4.1.*
2024-01-29T11:06:39.7709378Z  - yarp-matlab-bindings 3.9.0.*
2024-01-29T11:06:39.7709727Z  - yarp-device-openxrheadset 0.0.3.*
2024-01-29T11:06:39.7710103Z  - blocktest-yarp-plugins 1.1.4.*
2024-01-29T11:06:39.7710468Z  - robot-testing-framework 2.0.1.*
2024-01-29T11:06:39.7710823Z  - icub-firmware-shared 1.37.0.*
2024-01-29T11:06:39.7711142Z  - openxr-sdk 1.0.33.*
2024-01-29T11:06:39.7711445Z  - whole-body-controllers 2.5.6.*
2024-01-29T11:06:39.7711766Z  - wearables 1.8.0.*
2024-01-29T11:06:39.7712050Z  - idyntree-yarp-tools 0.0.9.*
2024-01-29T11:06:39.7712395Z  - liblie-group-controllers 0.2.0.*
2024-01-29T11:06:39.7712740Z  - pyqtconsole 1.2.3.*
2024-01-29T11:06:39.7713024Z  - human-gazebo 1.1.0.*
2024-01-29T11:06:39.7713297Z  - qpoases 3.2.1.*
2024-01-29T11:06:39.7713583Z  - walking-controllers 0.8.0.*
2024-01-29T11:06:39.7713911Z  - ergocub-software 0.6.0.*
2024-01-29T11:06:39.7714235Z  - robot-log-visualizer 0.8.0.*
2024-01-29T11:06:39.7714553Z  - blockfactory 0.8.4.*
2024-01-29T11:06:39.7714870Z  - idyntree-matlab-bindings 10.2.0.*
2024-01-29T11:06:39.7715207Z  - manif 0.0.4.*
2024-01-29T11:06:39.7715475Z  - meshcat-python 0.3.2.*
2024-01-29T11:06:39.7715768Z  - libyarp 3.9.0.*
2024-01-29T11:06:39.7716032Z  - osqp-eigen 0.8.1.*
2024-01-29T11:06:39.7716333Z  - icub-contrib-common 1.19.0.*
2024-01-29T11:06:39.7716687Z  - matlab-whole-body-simulator 3.3.0.*
2024-01-29T11:06:39.7717064Z  - whole-body-estimators 0.10.0.*
2024-01-29T11:06:39.7717388Z  - idyntree 10.2.0.*
2024-01-29T11:06:39.7717687Z  - casadi-matlab-bindings 3.6.3.0.*
2024-01-29T11:06:39.7718033Z  - icub-basic-demos 1.24.0.*
2024-01-29T11:06:39.7718351Z  - funny-things 2.2.0.*
2024-01-29T11:06:39.7718657Z  - libbayes-filters-lib 0.10.0.*
2024-01-29T11:06:39.7718975Z  - librobometry 1.2.2.*
2024-01-29T11:06:39.7719303Z  - libunicycle-footstep-planner 0.7.0.*
2024-01-29T11:06:39.7719660Z  - proxsuite 0.6.0.*
2024-01-29T11:06:39.7719969Z  - human-dynamics-estimation 3.0.0.*
2024-01-29T11:06:39.7720306Z  - event-driven 2.1.*
2024-01-29T11:06:39.7720581Z  - blocktest 2.3.7.*
2024-01-29T11:06:39.7720874Z  - libgazebo-yarp-plugins 4.9.0.*
2024-01-29T11:06:39.7721345Z  - casadi 3.6.3.*
2024-01-29T11:06:39.7721673Z  - osqp-matlab 0.6.2.3.*
2024-01-29T11:06:39.7722060Z  - cppad 20230000.0.*
2024-01-29T11:06:39.7722374Z  - resolve-robotics-uri-py 0.1.0.*
2024-01-29T11:06:39.7722721Z  - ycm-cmake-modules 0.16.2.*
2024-01-29T11:06:39.7723035Z  - libmatio-cpp 0.2.3.*
2024-01-29T11:06:39.7723208Z 
2024-01-29T11:06:39.7723292Z with channels:
2024-01-29T11:06:39.7723532Z  - local
2024-01-29T11:06:39.7723754Z  - conda-forge
2024-01-29T11:06:39.7723994Z  - robotology
2024-01-29T11:06:39.7724129Z 
2024-01-29T11:06:39.7724229Z The reported errors are:
2024-01-29T11:06:39.7724557Z - Encountered problems while solving:
2024-01-29T11:06:39.7725510Z -   - package walking-controllers-0.8.0-h59595ed_112 requires libbipedal-locomotion-framework >=0.18.0,<0.18.1.0a0, but none of the providers can be installed
2024-01-29T11:06:39.7726885Z -   - package casadi-matlab-bindings-3.6.3.0-h59595ed_112 requires casadi >=3.6.3,<3.7.0a0, but none of the providers can be installed
2024-01-29T11:06:39.7727592Z - 
2024-01-29T11:06:39.7727698Z 
2024-01-29T11:06:39.7727937Z During handling of the above exception, another exception occurred:
2024-01-29T11:06:39.7728294Z 
2024-01-29T11:06:39.7728400Z Traceback (most recent call last):
2024-01-29T11:06:39.7728998Z   File "/usr/share/miniconda3/envs/test/bin/conda-mambabuild", line 10, in <module>
2024-01-29T11:06:39.7729517Z     sys.exit(main())
2024-01-29T11:06:39.7729752Z              ^^^^^^
2024-01-29T11:06:39.7730412Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/boa/cli/mambabuild.py", line 256, in main
2024-01-29T11:06:39.7731125Z     call_conda_build(action, config)
2024-01-29T11:06:39.7731940Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/boa/cli/mambabuild.py", line 228, in call_conda_build
2024-01-29T11:06:39.7732654Z     result = api.build(
2024-01-29T11:06:39.7732896Z              ^^^^^^^^^^
2024-01-29T11:06:39.7733561Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/conda_build/api.py", line 254, in build
2024-01-29T11:06:39.7734194Z     return build_tree(
2024-01-29T11:06:39.7734441Z            ^^^^^^^^^^^
2024-01-29T11:06:39.7735132Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/conda_build/build.py", line 3789, in build_tree
2024-01-29T11:06:39.7735815Z     packages_from_this = build(
2024-01-29T11:06:39.7736109Z                          ^^^^^^
2024-01-29T11:06:39.7736823Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/conda_build/build.py", line 2573, in build
2024-01-29T11:06:39.7737512Z     create_build_envs(top_level_pkg, notest)
2024-01-29T11:06:39.7738343Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/conda_build/build.py", line 2407, in create_build_envs
2024-01-29T11:06:39.7739034Z     raise e
2024-01-29T11:06:39.7739734Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/conda_build/build.py", line 2378, in create_build_envs
2024-01-29T11:06:39.7740444Z     environ.get_install_actions(
2024-01-29T11:06:39.7741288Z   File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/boa/cli/mambabuild.py", line 150, in mamba_get_install_actions
2024-01-29T11:06:39.7742028Z     raise err
2024-01-29T11:06:39.7743706Z conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("walking-controllers==0.8.0=h59595ed_112"), MatchSpec("casadi-matlab-bindings==3.6.3.0=h59595ed_112"), MatchSpec("libbipedal-locomotion-framework[version='>=0.18.0,<0.18.1.0a0']"), MatchSpec("casadi[version='>=3.6.3,<3.7.0a0']")}
2024-01-29T11:06:39.7745838Z -   - package walking-controllers-0.8.0-h59595ed_112 requires libbipedal-locomotion-framework >=0.18.0,<0.18.1.0a0, but none of the providers can be installed
2024-01-29T11:06:39.7747126Z -   - package casadi-matlab-bindings-3.6.3.0-h59595ed_112 requires casadi >=3.6.3,<3.7.0a0, but none of the providers can be installed
2024-01-29T11:06:39.7747819Z - 
2024-01-29T11:06:39.7747935Z 
2024-01-29T11:06:39.7748040Z Leaving build/test directories:
2024-01-29T11:06:39.7748323Z   Work:
2024-01-29T11:06:39.7748707Z  /usr/share/miniconda3/envs/test/conda-bld/work 
2024-01-29T11:06:39.7749062Z   Test:
2024-01-29T11:06:39.7749389Z  /usr/share/miniconda3/envs/test/conda-bld/test_tmp 
2024-01-29T11:06:39.7749786Z Leaving build/test environments:
2024-01-29T11:06:39.7750067Z   Test:
2024-01-29T11:06:39.7751387Z source activate  /usr/share/miniconda3/envs/test/conda-bld/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p 
2024-01-29T11:06:39.7752615Z   Build:
2024-01-29T11:06:39.7753031Z source activate  /usr/share/miniconda3/envs/test/conda-bld/_build_env 
2024-01-29T11:06:39.7753435Z 
2024-01-29T11:06:39.7753440Z 
2024-01-29T11:06:39.8954104Z ##[error]Process completed with exit code 1.
2024-01-29T11:06:39.9038272Z Post job cleanup.
2024-01-29T11:06:40.0815210Z [command]/usr/bin/git version
2024-01-29T11:06:40.0853226Z git version 2.43.0
2024-01-29T11:06:40.0903468Z Temporarily overriding HOME='/home/runner/work/_temp/a7c7954d-0c05-4d7a-a2b2-507db90026b9' before making global git config changes
2024-01-29T11:06:40.0905128Z Adding repository directory to the temporary git global config as a safe directory
2024-01-29T11:06:40.0908985Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/robotology-superbuild/robotology-superbuild
2024-01-29T11:06:40.0933054Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-01-29T11:06:40.0954601Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-01-29T11:06:40.1107775Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-01-29T11:06:40.1119711Z http.https://github.com/.extraheader
2024-01-29T11:06:40.1131035Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-01-29T11:06:40.1151013Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-01-29T11:06:40.1513481Z Cleaning up orphan processes
~~~

**Comment by traversaro** [comment]
Trying to solve the env:
~~~
name: testenv
channels:
  - conda-forge
  - robotology
dependencies:
    - ycm-cmake-modules=0.16.2
    - robot-testing-framework=2.0.1
    - libyarp=3.9.0
    - icub-firmware-shared=1.37.0
    - icub-main=2.4.1
    - icub-contrib-common=1.19.0
    - icub-models=2.4.1
    - ergocub-software=0.6.0
    - robots-configuration=2.5.2
    - libgazebo-yarp-plugins=4.9.0
    - yarp-matlab-bindings=3.9.0
    - icub-tests=1.25.0
    - blocktest=2.3.7
    - blocktest-yarp-plugins=1.1.4
    - libosqp=0.6.3
    - osqp-eigen=0.8.1
    - idyntree=10.2.0
    - idyntree-matlab-bindings=10.2.0
    - idyntree-yarp-tools=0.0.9
    - qpoases=3.2.1
    - blockfactory=0.8.4
    - wb-toolbox=5.6.0
    - libunicycle-footstep-planner=0.7.0
    - whole-body-estimators=0.10.0
    - whole-body-controllers=2.5.6
    - libmatio-cpp=0.2.3
    - librobometry=1.2.2
    - osqp-matlab=0.6.2.3
    - matlab-whole-body-simulator=3.3.0
    - libbayes-filters-lib=0.10.0
    - manif=0.0.4
    - proxsuite=0.6.0
    - casadi=3.6.3
    - cppad=20230000.0
    - liblie-group-controllers=0.2.0
    - bipedal-locomotion-framework=0.18.0
    - walking-controllers=0.8.0
    - casadi-matlab-bindings=3.6.3.0
    - pyqtconsole=1.2.3
    - meshcat-python=0.3.2
    - robot-log-visualizer=0.8.0
    - resolve-robotics-uri-py=0.1.0
    - wearables=1.8.0
    - human-dynamics-estimation=3.0.0
    - walking-teleoperation=1.3.4
    - openxr-sdk=1.0.33
    - yarp-device-openxrheadset=0.0.3
    - openvr=1.16.8.1
    - yarp-openvr-trackers=0.0.1
    - yarp-devices-forcetorque=0.3.1
    - human-gazebo=1.1.0
    - event-driven=2.1
    - yarp-device-xsensmt=0.3.0
    - icub-basic-demos=1.24.0
    - speech=1.2.0
    - funny-things=2.2.0
~~~

locally I have a more clear error:
~~~
Could not solve for environment specs
The following packages are incompatible
├─ bipedal-locomotion-framework 0.18.0**  is installable and it requires
│  └─ libbipedal-locomotion-framework [0.18.0 ha482fe9_1|0.18.0 he4f1b30_0], which requires
│     └─ cppad >=20240000.2,<20240000.3.0a0 , which can be installed;
├─ casadi 3.6.3**  is installable with the potential options
│  ├─ casadi 3.6.3 would require
│  │  └─ proxsuite >=0.5.1,<0.6.0a0  with the potential options
│  │     ├─ proxsuite 0.5.1, which can be installed;
│  │     └─ proxsuite [0.5.1|0.6.0|0.6.1|0.6.2|0.6.3] would require
│  │        └─ python_abi 3.12.* *_cp312, which can be installed;
│  ├─ casadi 3.6.3 would require
│  │  └─ proxsuite >=0.3.7,<0.4.0a0 , which can be installed;
│  ├─ casadi 3.6.3 would require
│  │  └─ proxsuite >=0.6.1,<0.7.0a0  with the potential options
│  │     ├─ proxsuite [0.5.1|0.6.0|0.6.1|0.6.2|0.6.3], which can be installed (as previously explained);
│  │     └─ proxsuite [0.6.1|0.6.2|0.6.3], which can be installed;
│  ├─ casadi 3.6.3 would require
│  │  └─ proxsuite >=0.5.0,<0.6.0a0  with the potential options
│  │     ├─ proxsuite 0.5.0, which can be installed;
│  │     ├─ proxsuite 0.5.1, which can be installed;
│  │     └─ proxsuite [0.5.1|0.6.0|0.6.1|0.6.2|0.6.3], which can be installed (as previously explained);
│  └─ casadi 3.6.3 would require
│     └─ python_abi 3.12.* *_cp312, which can be installed;
├─ cppad 20230000.0**  is not installable because it conflicts with any installable versions previously reported;
├─ proxsuite 0.6.0**  is installable with the potential options
│  ├─ proxsuite [0.5.1|0.6.0|0.6.1|0.6.2|0.6.3], which can be installed (as previously explained);
│  └─ proxsuite 0.6.0 conflicts with any installable versions previously reported;
├─ walking-controllers 0.8.0**  does not exist (perhaps a typo or a missing channel);
└─ wearables 1.8.0**  is not installable because there are no viable options
   ├─ wearables 1.8.0 would require
   │  └─ python_abi 3.10.* *_cp310, which conflicts with any installable versions previously reported;
   ├─ wearables 1.8.0 would require
   │  └─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported;
   ├─ wearables 1.8.0 would require
   │  └─ python_abi 3.8.* *_cp38, which conflicts with any installable versions previously reported;
   └─ wearables 1.8.0 would require
      └─ python_abi 3.9.* *_cp39, which conflicts with any installable versions previously reported.
~~~

In a nutshell, for sure we need to bump at least cppad

**Comment by traversaro** [comment]
> the problems goes away removing the proxsuite pin, so probably it is just a missing rebuilt of casadi with the latest proxsuite.

Actually it is again a problem on our side, proxsuite should be updated to 0.6.1 .

**Comment by traversaro** [comment]
While indeed the lagging behind of proxsuite and cppad could be a bit a problem in robotology-superbuild and a bit a lack of pinning for these two libraries in conda-forge-pinning, in general we do not care too much about the version of these libraries, even if they are tracked in latest.releases.yaml . Probably we could consider to just avoid adding them to the version listed in robotology-distro and robotology-distro-all .

**Comment by traversaro** [comment]
> > `generate-conda-packages` test build with these modifications : https://github.com/robotology/robotology-superbuild/actions/runs/7693316196
> 
> new test build: https://github.com/robotology/robotology-superbuild/actions/runs/7695901115 .

Build successful, let's merge.
